locale.c

```diff
@@ -10090,7 +10090,7 @@ Perl_mem_collxfrm_(pTHX_ const char *input_string,
     if (UNLIKELY(! xbuf)) {
         DEBUG_L(PerlIO_printf(Perl_debug_log,
                       "mem_collxfrm_: Couldn't malloc %zu bytes\n", xAlloc));
-        goto bad;
+        goto bad_no_strxfrm;
```

Comment:
This looks like a potential memory leak to me. At this point, `sans_highs`, `sans_nuls`, and `free_me` may have been allocated, but `bad_no_strxfrm` skips over the `free()` calls.

Reply:
Somehow I got it into my head that that label had code to do the necessary cleanup, and was blind to the fact that it didn't.  But with the latest push, it should now.